spring-cloud-sleuth-brave/src/main/java/org/springframework/cloud/sleuth/brave/instrument/web/BraveHttpConfiguration.java

```diff
@@ -0,0 +1,291 @@
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.brave.instrument.web;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.validation.constraints.NotNull;
+
+import brave.Tracing;
+import brave.http.HttpRequest;
+import brave.http.HttpTracing;
+import brave.http.HttpTracingCustomizer;
+import brave.sampler.SamplerFunction;
+import brave.sampler.SamplerFunctions;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.sleuth.api.http.HttpRequestParser;
+import org.springframework.cloud.sleuth.api.http.HttpResponseParser;
+import org.springframework.cloud.sleuth.brave.bridge.BraveHttpBridgeConfiguration;
+import org.springframework.cloud.sleuth.brave.bridge.BraveHttpRequestParser;
+import org.springframework.cloud.sleuth.brave.bridge.BraveHttpResponseParser;
+import org.springframework.cloud.sleuth.brave.bridge.BraveSamplerFunction;
+import org.springframework.cloud.sleuth.instrument.web.HttpClientRequestParser;
+import org.springframework.cloud.sleuth.instrument.web.HttpClientResponseParser;
+import org.springframework.cloud.sleuth.instrument.web.HttpClientSampler;
+import org.springframework.cloud.sleuth.instrument.web.HttpServerRequestParser;
+import org.springframework.cloud.sleuth.instrument.web.HttpServerResponseParser;
+import org.springframework.cloud.sleuth.instrument.web.HttpServerSampler;
+import org.springframework.cloud.sleuth.instrument.web.SkipPatternProvider;
+import org.springframework.cloud.sleuth.instrument.web.SleuthHttpProperties;
+import org.springframework.cloud.sleuth.instrument.web.SleuthWebProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.lang.Nullable;
+
+/**
+ * {@link org.springframework.boot.autoconfigure.EnableAutoConfiguration
+ * Auto-configuration} related to HTTP based communication.
+ *
+ * @author Marcin Grzejszczak
+ * @since 2.0.0
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnProperty(name = { "spring.sleuth.http.enabled", "spring.sleuth.web.enabled" }, havingValue = "true",
+		matchIfMissing = true)
+@ConditionalOnClass(HttpTracing.class)
+@EnableConfigurationProperties({ SleuthWebProperties.class, SleuthHttpProperties.class })
+public class BraveHttpConfiguration {
+
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnBean(Tracing.class)
+	@Import(BraveHttpBridgeConfiguration.class)
```

Comment:
This configuration class is imported only here. Why not merge the two beans in this configuration. This'll avoid exposing a public class that you may import several times by accident.